src/item.h

```diff
@@ -1915,13 +1915,18 @@ class item : public visitable<item>
         inline void set_owner( faction_id new_owner ) {
             owner = new_owner;
         }
+        void set_owner( player &p );
+        void set_owner( Character &c );
```

Comment:
I think you just need the Character version; if you pass `player &p` to the Character version, it will be classed correctly.